src/main/kotlin/snyk/container/YAMLImageExtractor.kt

```diff
@@ -29,16 +29,19 @@ object YAMLImageExtractor {
     }
 
     private fun extractImageNameFromLine(s: String): String {
-        return s.trim()
-            .split(":")
-            .stream().filter { e -> e.trim() != "image" }
-            .map { image -> image.trim() }
-            .collect(Collectors.joining(":"))
+        return if (!s.trim().startsWith("#")) {
+            imageRegEx.find(s)?.value?.trim() ?: ""
+        } else {
+            ""
+        }
     }
 
     fun extractFromFile(file: VirtualFile, project: Project): Set<KubernetesWorkloadImage> {
         val psiFile = findPsiFileIgnoringExceptions(file, project)
-        if (psiFile == null || file.isDirectory || !file.isValid) return emptySet()
+        if (psiFile == null || file.isDirectory || !file.isValid) {
+            logger.warn("no psi file found for ${file.path}")
```

Comment:
will produce too many warns for folders/dirs
PS also for any deleted/in-memory file, I guess